packages/data-migration/src/cli/InteractiveCliStatusReporter.ts

```diff
@@ -0,0 +1,59 @@
+import readline from "readline";
+import { MigrationStatusReporter } from "~/cli/MigrationStatusReporter";
+import { MigrationStatus } from "~/types";
+import { LogReporter } from "~/cli/LogReporter";
+
+export class InteractiveCliStatusReporter implements MigrationStatusReporter {
+    private logReporter: LogReporter;
+
+    constructor(logReporter: LogReporter) {
+        this.logReporter = logReporter;
+        console.log(`Using "InteractiveCliStatusReporter".`);
+    }
+
+    report(migrationStatus: MigrationStatus): void {
+        const { status, migrations, context } = migrationStatus;
+        this.clearLine();
+
+        const currentLogStreamName = context?.logStreamName;
+        if (currentLogStreamName) {
+            this.logReporter.initializeStream(currentLogStreamName);
+        }
+
+        if (status === "running") {
```

Comment:
These ifs are good candicates for for a switch statement, with a default behavior when status is not one of the running, init or done.

Reply:
This is just a reporter, meaning, it prints stuff to CLI. Here I'm only interested in these statuses, others I just ignore. As for `switch`, I have a better idea there, but for this release I won't spend any more time on making this any more advanced than what it is.